apps/rush-lib/src/cli/actions/BaseInstallAction.ts

```diff
@@ -39,6 +40,11 @@ export abstract class BaseInstallAction extends BaseRushAction {
         + ' This flag is useful for automated builds that want to report stages individually'
         + ' or perform extra operations in between the two stages.'
     });
+    this._debugPackageManagerParameter = this.defineFlagParameter({
+      parameterLongName: '--debug-package-manager',
+      description: 'Activates verbose logging for the package manager. You will probably want to pipe'
+        + ' the output of Rush to a file when using this command.'
```

Comment:
It would be better for `--debug-package-manager` to create this file itself.  Since this switch will be primarily used when reporting bugs to the Rush developers, we'd rather get the diagnostic files in a standardized format versus each person making their own attempt to collect the output.

Maybe consider an approach like this:

1. You run `rush --collect-logs`
2. You zip up `common/temp/rush-logs/*`
3. You email it to us.

The command erases `common/temp/rush-logs/` at the start of the operation, and then copies diagnostic in there.  In the beginning it can be just the PNPM log file, but eventually it could include:

- the Rush verbose log output
- a copy of the shrinkwrap file (which Zoltan usually asks for)
- a file with some PC stats such as OS version, NodeJS, etc.

Reply:
Right, but without re-writing this code to have a bunch of async stream handling stuff in it, it is not possible to stream the logs to stdout while also capturing them in a file. I thought it significantly increased the complexity of the code.

I would be in favor of doing that in another PR, especially once we started using an actual logging library (e.g. not writing console.log everywhere).

Reply:
I'm cool with that.

---
In reply to: [203510280](https://github.com/Microsoft/web-build-tools/pull/745#discussion_r203510280) [](ancestors = 203510280)